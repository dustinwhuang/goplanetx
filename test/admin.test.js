import React from 'react';
import AdminView from '../client/src/components/admin/adminView.jsx';
import Message from '../client/src/components/admin/message.jsx';
import Response from '../client/src/components/admin/response.jsx';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

/*TEST ADMIN COMPONENTS*/

// describe('Component: Message', () => {
//   const message = {
//     createdAt: '2017-11-25T15:13:29.000Z',
//     first_name: 'Brad',
//     last_name: 'Pitt',
//     user_contact: 'the quick brown fox',
//     user_message: 'jumped over the lazy dog',
//     user_urgency: '2'
//   };

//   it('should match its empty snapshot', () => {
//     const component = shallow( <Message message={message}/>);

//     expect(component).toMatchSnapshot();
//   });

// });

describe('Component: AdminView', () => {
  const showLogIn = () => {
    console.log('show log in');
  };
  const retrieveOpenMessages = () => {
    console.log('retrieve open messages');
  };
  
  it('should match its empty snapshot', () => {
    const component = shallow( <AdminView showLogIn={showLogIn} retrieveOpenMessages={retrieveOpenMessages}/>);

    expect(component).toMatchSnapshot();
  });

});

describe('Component: Response', () => {
  const messageName = 'Brad Pitt';
  const messageId = 2;
  const submitAdminResponse = () => {
    console.log('submit admin response');
  };
  
  it('should match its empty snapshot', () => {
    const component = shallow( <Response messageName={messageName} messageId={messageId} submitAdminResponse={submitAdminResponse}/>);

    expect(component).toMatchSnapshot();
  });

});
