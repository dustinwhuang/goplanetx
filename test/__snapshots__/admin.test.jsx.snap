// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AdminView should match its empty snapshot 1`] = `
<div>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="primary"
    className="admin-change-user-button"
    disabled={false}
    onClick={[Function]}
  >
    Sign In as a Different User
  </Button>
  <div
    className="admin-header group"
  >
    <h3
      className="welcome-header"
    >
      Welcome to Your Inbox!
    </h3>
    <h4>
      You can view and respond to user messages here.
    </h4>
  </div>
  <ul
    className="user-message-ul"
  >
    <Message
      key="1"
      message={
        Object {
          "createdAt": 2017-11-30T00:04:40.750Z,
          "first_name": "Jane",
          "id": 1,
          "last_name": "Smith",
          "user_contact": "Test contact info",
          "user_message": "Test message",
          "user_urgency": 3,
        }
      }
      setResponseId={[Function]}
      submitAdminResponse={[Function]}
    />
    <Message
      key="2"
      message={
        Object {
          "createdAt": 2017-11-30T00:04:40.750Z,
          "first_name": "Lady",
          "id": 2,
          "last_name": "Person",
          "user_contact": "Test contact info 2",
          "user_message": "Test message 2",
          "user_urgency": 1,
        }
      }
      setResponseId={[Function]}
      submitAdminResponse={[Function]}
    />
  </ul>
</div>
`;

exports[`Component: Response should match its empty snapshot 1`] = `
<FormGroup
  bsClass="form-group"
>
  <div>
    Respond to 
    Brad Pitt
    's message:
  </div>
  <FormControl
    bsClass="form-control"
    componentClass="textarea"
    onChange={[Function]}
    placeholder="Response..."
    type="text"
  />
  <br />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    disabled={false}
    onClick={[Function]}
  >
    Submit Response
  </Button>
</FormGroup>
`;
